test/IISIntegration.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -0,0 +1,173 @@
+
+        public IISApplication(DeploymentParameters deploymentParameters, ILogger logger)
+        {
+            _deploymentParameters = deploymentParameters;
+            _logger = logger;
+            WebSiteName = CreateTestSiteName();
+            AppPoolName = $"{WebSiteName}Pool";
+        }
+
+        public string WebSiteName { get; }
+
+        public string AppPoolName { get; }
+
+        public void StartIIS(Uri uri, string contentRoot)
+        {
+            using (_logger.BeginScope("StartIIS"))
+            {
+                var port = uri.Port;
+                if (port == 0)
+                {
+                    port = (uri.Scheme == "https") ? TestPortHelper.GetNextSSLPort() : TestPortHelper.GetNextPort();
+                }
+
+                ConfigureAppPool();
+
+                ConfigureSite(contentRoot, port);
+
+                ConfigureAppHostConfig(contentRoot);
+
+                _serverManager.CommitChanges();
+            }
+        }
+
+        private void ConfigureAppPool()
+        {
+            var pool = _serverManager.ApplicationPools.Add(AppPoolName);
+            pool.StartMode = StartMode.AlwaysRunning;
+            pool.ProcessModel.IdentityType = ProcessModelIdentityType.LocalSystem;
+            pool.ProcessModel.LoadUserProfile = true;
+        }
+
+        private void ConfigureSite(string contentRoot, int port)
+        {
+            var site = _serverManager.Sites.Add(WebSiteName, contentRoot, port);
+            site.Applications.Single().ApplicationPoolName = AppPoolName;
+        }
+
+        private void ConfigureAppHostConfig(string dllRoot)
+        {
+            var config = _serverManager.GetApplicationHostConfiguration();
+            var globalModulesSection = config.GetSection("system.webServer/globalModules");
+
+            var ancmFile = GetAncmLocation(dllRoot);
+            var found = false;
+            foreach (var element in globalModulesSection.GetCollection())
+            {
+                if ((string)element["name"] == @"AspNetCoreModuleV2")
+                {
+                    _originalAspNetCoreDllLocation = (string)element["image"];
+                    element["image"] = ancmFile;
+                    found = true;
+                    break;
+                }
+            }
+
+            // If ANCMV2 isn't in the global apphost.config, add it there.
+            if (!found)
+            {
+                _originalAspNetCoreDllLocation = "";
+                var addElement = globalModulesSection.GetCollection().CreateElement("add");
+                addElement["name"] = "AspNetCoreModuleV2";
+                addElement["image"] = ancmFile;
+                globalModulesSection.GetCollection().Add(addElement);
+            }
+        }
+
+        private string GetAncmLocation(string dllRoot)
+        {
+            var arch = _deploymentParameters.RuntimeArchitecture == RuntimeArchitecture.x64 ? @"x64\aspnetcorev2.dll" : @"x86\aspnetcorev2.dll";
```

Comment:
We may have to find a way to prevent any IIS parallel test execution, even across repos.

Reply:
I'd handle this in another PR as this is the only repo doing it.

Reply:
Across repos is a non-goal for right now.